dialects/tests/test1.yaml

```diff
@@ -0,0 +1,35 @@
+name: testdb
+dependencies:
+  arithmetic: 
+    https://github.com/substrait-io/substrait/blob/main/extensions/functions_arithmetic.yaml
+  comparison: 
+    https://github.com/substrait-io/substrait/blob/main/extensions/functions_comparison.yaml
+supported_types:
+  i32:
+    sql_type_name: TINYINT
+    supported_as_column: true
+  i64:
+    sql_type_name: TINYINT
+    supported_as_column: true
+scalar_functions:
+- name: comparison.is_null
+  local_name: IS NULL
+  notation: POSTFIX
```

Comment:
what is the point of these? they seem like they relate to how this plan would be written as sql, but why is that relevant here?

Reply:
These aid in conversion to and from a specific SQL dialect.  They are a feature of the existing dialect files supported in substrait-go.

Reply:
Interesting, so kinda like defining the consumer directly in the dialect yaml, I guess that could be nice. How far do we want to take it though? Sometimes the target functions take the same args but in different order, should we allow for e.g. that as well?

Nb. maybe the "sql_type_name" above should be "local_name" as well?

Reply:
Done.

Reply:
(Order difference is likely a different function implementation.)